draft-ietf-quic-recovery.md

```diff
@@ -1086,8 +1086,9 @@ the congestion window SHOULD NOT be increased in either slow start or
 congestion avoidance. This can happen due to insufficient application data
 or flow control limits.
 
-A sender can use the pipeACK method described in Section 4.3 of {{?RFC7661}} to
-determine if the congestion window is sufficiently utilized.
+A sender can use a variety of mechnaisms to determine if the congestion window
+is sufficiently utuilized.  For example, the pipeACK method described in
+Section 4.3 of {{?RFC7661}}.
```

Comment:
I would actually rephrase this part completely (or maybe just remove it entirely as there is another reference two paragraphs down). RFC7661 mainly uses the term "application-limited", however, it does not really propose a mechanism to exactly _detect_ that a connection is application-limited, but rather proposes additional mechanism to even decrease the congestion window when no validated. It further says that the cwnd window MUST NOT be increase when not cwnd-limited and that cwnd-limitation is detected based on FlightSize (or other mechanism, see also section 4.5.3 of RFC7661) as also described in the previous paragraph in the recovery draft:

```
A sender that is not cwnd-limited MUST NOT increase the cwnd
         when ACK packets are received in this phase 
```
So I don't think RFC7661 actually provides any different meachanism to detect if the cwnd is sufficiently utilised.

Please note also that RFC7661 actually says MUST NOT while the QUIC recovery draft has only a SHOULD NOT...

Reply:
Good point, I'll remove this paragraph.